misc/python/materialize/cli/mzbench.py

```diff
@@ -0,0 +1,160 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# mzbuild.py -- script to run materialized benchmarks
+
+import argparse
+import csv
+import itertools
+import os
+import subprocess
+import sys
+import typing
+
+import psutil  # type: ignore
+
+
+def rev_parse(git_ref: str) -> str:
+    if not git_ref:
+        return git_ref
+    return subprocess.check_output(["git", "rev-parse", git_ref]).strip().decode()
+
+
+def main(
+    composition: str,
+    worker_counts: typing.List[int],
+    git_revisions: typing.List[typing.Union[str, None]],
+) -> None:
+
+    # This includes hypercores
+    physical_cpus = psutil.cpu_count(logical=False)
+    if physical_cpus < 4:
+        # Explicitly override the worker counts for the CI benchmark
+        benchmark = "benchmark-ci"
+        worker_counts = [1]
+    elif physical_cpus < 8:
+        benchmark = "benchmark-medium"
+    else:
+        benchmark = "benchmark"
```

Comment:
This seems weirdly implicit to me! Kind of seems like something you'd want the user to explicitly specify when running `mzbench`?

Reply:
We can make it either implicit or explicit. The idea is that we curate the benchmark, and the size of each, based on the size of the machine you're running on. We could default to `benchmark-medium` and then add a flag to specify an alternate benchmark if you prefer making this explicit.

Typing that out, it seems reasonable to make this explicit for the cloud based benchmarks.